desktop-widgets/divelogexportdialog.cpp

```diff
@@ -131,6 +131,9 @@ void DiveLogExportDialog::on_exportGroup_buttonClicked(QAbstractButton*)
 	showExplanation();
 }
 
+// We use one future for all exports. This is no problem as we also have only one notification widget.
+static QFuture<void> future;
```

Comment:
can this be an member of the dialog instance?

Reply:
I think so, yes. However, KMessageWidget is "part of the KDE libraries" according to the file header, so I didn't want to touch it.

Reply:
Oh - I think I misunderstood - you suggest to make it a member of DiveLogExportDialog, not of KMessageWidget. Sure, this can be done.

Reply:
Stupid question: Do we even have to keep a QFuture object? Once we've passed it down to the notification widget - can't we destroy the original? I'm not familiar with copy-semantics of QFuture (I assume it is copied in the QFutureWatcher).

Reply:
i do not know for sure. examples like this one are keeping it around as a class member: http://codejourneys.blogspot.com/2008/06/qt-simple-example-of-use-of.html

Reply:
This example has a number of problems - new but no delete...

According to the Qt docs, QFuture watcher is designed to be passed around as value:
```
QFuture is a lightweight reference counted class that can be passed by value.
```
I tested it on a huge log, and it worked flawlessly. Let's remove that global variable.